### ISSUE TITLE
Difference in build success across branches?

### ISSUE THREAD

**Issue by doctor-phil** [QuantEcon/lecture-datascience.myst#243]
It seems like very frequently, test/preview builds will succeed on alternative branches, but fail when they are merged into `main`. 

Furthermore, when drafting a release/publish from main, more build issues will appear even if the preview build succeeds after the latest change on `main`

This causes a 3+ step debug routine for any changes that is not ideal. @mmcky do you have any idea what might be causing this disconnect?

**Comment by mmcky** [comment]
@doctor-phil thanks for the heads up. We don't experience those issues. I will take a look at the workflows and see what might be causing the issues. 

Some things to check:

- [ ] Is the same environment being used for all build workflows?
- [ ] Is a build cache implemented for merges to `main` to speed up PRs

**Comment by mmcky** [comment]
@doctor-phil it looks like many of the preview build failures are due to a new version of pandas or matplotlib with code internal to the lecture requiring updating

```
'Legend' object has no attribute 'legendHandles'
```

**Comment by doctor-phil** [comment]
@mmcky yeah that's why I made the environment so restrictive. Without pinning the specific versions, every update causes some of the lectures to break. In this specific case, the preview builds on PR #242 succeeded on the branch, but failed the preview build when merged to `main` because of a typo in the changes.

Then once I got the preview build working on `main` I drafted a publish/release and now there are a bunch of errors in the lecture code that weren't caught in either of the previous preview builds

**Comment by mmcky** [comment]
> @mmcky yeah that's why I made the environment so restrictive. Without pinning the specific versions, every update causes some of the lectures to break. In this specific case, the preview builds on PR #242 succeeded on the branch, but failed the preview build when merged to `main` because of a typo in the changes.
> 
> Then once I got the preview build working on `main` I drafted a publish/release and now there are a bunch of errors in the lecture code that weren't caught in either of the previous preview builds

@doctor-phil one way this can happen is the `CI` uses a build cache to increase performance so only the lecture(s) that are changing get executed. When it runs on the `main` branch there is a fresh `cache` run done so if any software has changed it can cause failures if the existing lectures have issues re: versions. Have you updated any software versions?

**Comment by mmcky** [comment]
Also re: https://github.com/QuantEcon/lecture-datascience.myst/pull/242 it doesn't look like this actually was built on the PR as it originated from a fork. Unfortunately when a PR originates from a fork GitHub requires you to authorise the run (and sometimes that even isn't available). did you see a netlify preview link?

**Comment by doctor-phil** [comment]
@mmcky I haven't updated any software versions (intentionally). I should have time the next couple days to check out the current build errors since that might give some more insights.

Also, I didn't realize that it didn't do the preview build for PR from a fork. I just saw the green check and assumed. 

Still though, seems odd that once it was merged to `main` I got the preview link successfully but failed the release build
